app/controllers/application_controller.rb

```diff
@@ -45,6 +45,21 @@ def user_not_authorized
   end
 
   def pundit_user
-    PunditContext.new(current_user, { current_project: @project, current_story: @story })
+    PunditContext.new(current_team, current_user, { current_project: @project, current_story: @story })
+  end
+
+  def current_team
+    raise ActiveRecord::RecordNotFound, 'Team not set' unless session[:current_team_slug]
+    @current_team ||= Team.not_archived.find_by_slug(session[:current_team_slug])
```

Comment:
question: what happens if the team was archived? it will return `nil` and what next?
I've checked the `PuditContext` and I didn't found any logic about it